contracts/src/v0.8/feeds/DualAggregator.sol

```diff
@@ -680,18 +716,53 @@ contract DualAggregator is OCR2Abstract, OwnerIsCreator, AggregatorV2V3Interface
     bytes32[] calldata ss,
     bytes32 rawVs
   ) external override {
-    // TODO: update this function with implementation from the doc
-
     // NOTE: If the arguments to this function are changed, _requireExpectedMsgDataLength and/or
     // TRANSMIT_MSGDATA_CONSTANT_LENGTH_COMPONENT need to be changed accordingly
 
     uint256 initialGas = gasleft(); // This line must come first
+    Report memory report_;
 
-    HotVars memory hotVars = s_hotVars;
-
+    if (msg.sender == s_secondaryProxy) {
+      // Decode the report
+      report_ = _decodeReport(report);
+      (bool exist, uint32 roundId) = _reportExists(report_);
+      // In case the report exists, only copy the round id
+      if (exist) {
+        s_hotVars.latestSecondaryRoundId = roundId;
+        return;
```

Comment:
I believe we still want to pay the transmitter even when a report already exists.

Reply:
I added the `_payTransmitter()` inside this case too.